### ISSUE TITLE
build failures

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#1917]
We have build failures on the python 3.9 conda builds (our setup makes it hard to find out if there's any other failures).

We can get numpy being built, leading to problems. We can appear to fix this in #1916 but there are still problems related to the `stk.so` library.

Is this related to the much-earlier apparent error:

```
    checking for python3.9... /usr/share/miniconda/envs/build/bin/python3.9
    checking for a version of Python >= '2.1.0'... yes
    checking for a version of Python >='2.7'... yes
    checking for the sysconfig Python package... yes
    checking for Python include path... -I/usr/share/miniconda/envs/build/include/pypy3.9
    checking for Python library path... -L/usr/share/miniconda/envs/build/lib -lpython3.9
    checking for Python site-packages path... /usr/share/miniconda/envs/build/lib/pypy3.9/site-packages
    checking for Python platform specific site-packages path...
    checking python extra libraries... Traceback (most recent call last):
      File "<string>", line 1, in <module>
    TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'

    checking python extra linking flags... None
    checking for python... /usr/share/miniconda/envs/build/bin/python3.9
    checking for numpy includedir... /tmp/pip-build-env-2wi_mj20/overlay/lib/pypy3.9/site-packages/numpy/core/include

    checking for x86_64-conda-linux-gnu-gcc... (cached) /usr/share/miniconda/envs/build/bin/x86_64-conda-linux-gnu-cc
```

which is from

```
=== configuring in grplib-4.9 (/home/runner/work/sherpa/sherpa/extern/grplib-4.9)
```

- for however long https://github.com/sherpa/sherpa/actions/runs/6382629959/job/17321647417?pr=1608#step:8:396 will list

**Comment by DougBurke** [comment]
Given this error is from `configure`, is this an `autoconf`-like issue?

**Comment by Marie-Terrell** [comment]
It might be due to the usage of "+" with print [here](https://github.com/autoconf-archive/autoconf-archive/blob/master/m4/ax_python_devel.m4)?

**Comment by DougBurke** [comment]
So I think it's from something like the following

```
printf %s "checking python extra libraries... " >&6; }
	if test -z "$PYTHON_EXTRA_LIBS"; then
	   PYTHON_EXTRA_LIBS=`$PYTHON -c "$IMPORT_SYSCONFIG; \
                conf = sysconfig.get_config_var; \
                print (conf('LIBS') + ' ' + conf('SYSLIBS'))"`
	fi
```

and for some reason the `get_config_var` call is returning None rather than a string for one (or both) options. But as to who, why, what, where I have no idea, or whether this is something different than the original problem. Or ...

**Comment by Marie-Terrell** [comment]
Alright. It looks like you can reproduce the issue by:
```
$> conda create -n build -c xspec/label/test python=3.9 gcc_linux-64 gxx_linux-64 gfortran_linux-64 numpy cython astropy matplotlib=3 xspec-modelsonly=12.12.0g setuptools'<60'
$> python -c "import sysconfig; conf = sysconfig.get_config_var; print (conf('LIBS') + ' ' + conf('SYSLIBS'))"
Traceback (most recent call last):                                                                                                                                                      
  File "<string>", line 1, in <module>                                                                                                                                                  
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
$> python -c "import sysconfig; conf = sysconfig.get_config_var; print(str(conf('LIBS')) + ' ' + str(conf('SYSLIBS')))"
None None
```

This fixes the issue:
```
$> conda create -n build -c xspec/label/test python="=3.9=*cpython*" gcc_linux-64 gxx_linux-64 gfortran_linux-64 numpy cython astropy matplotlib=3 xspec-modelsonly=12.12.0g setuptools'<60'
$> python -c "import sysconfig; conf = sysconfig.get_config_var; print (conf('LIBS') + ' ' + conf('SYSLIBS'))"
-lcrypt -lpthread -ldl  -lutil -lrt -lm -lm
```

It turns out the pypy build of Python is being installed in some cases and that installing the cpython version resolves the issue. It looks like the pypy package is the latest release here: https://anaconda.org/conda-forge/python/files?page=1

So, it looks like an update to AX_PYTHON_DEVEL.m4 is not needed, but maybe some restrictions to cpython? Or, if you expect pypy implementation to work, we may need to resolve an underlying issue here

**Comment by DougBurke** [comment]
I'd vote for the simplest,  minimal change for now!

Thanks for the investigation

**Comment by hamogu** [comment]
Agree with Doug. I've never used pypy and I don't think many of our users do. Let's just declare that out of scope for 4.16. However, there is no harm to keep that in mind and make it work with pypy eventually and then add a dedicated CI job for pypy instead of randomly sometimes triggering one and sometimes the other.

**Comment by DougBurke** [comment]
This has been fixed by #1918 so closing
